detectable/src/main/java/com/synopsys/integration/detectable/detectables/pnpm/lockfile/process/PnpmYamlTransformerv6.java

```diff
@@ -0,0 +1,201 @@
+        buildGraph(dependencyGraph, rootPackageIds, packageMap, linkedPackageResolver, reportingProjectPackagePath);
+
+        if (projectNameVersion != null) {
+            return new CodeLocation(
+                dependencyGraph,
+                ExternalId.FACTORY.createNameVersionExternalId(Forge.NPMJS, projectNameVersion.getName(), projectNameVersion.getVersion()),
+                sourcePath
+            );
+        }
+        return new CodeLocation(dependencyGraph, sourcePath);
+    }
+
+    private void buildGraph(
+        DependencyGraph graphBuilder,
+        List<String> rootPackageIds,
+        @Nullable Map<String, PnpmPackageInfov6> packageMap,
+        PnpmLinkedPackageResolver linkedPackageResolver,
+        @Nullable String reportingProjectPackagePath
+    ) throws IntegrationException {
+        if (packageMap == null) {
+            throw new DetectableException("Could not parse 'packages' section of the pnpm-lock.yaml file.");
+        }
+        for (Map.Entry<String, PnpmPackageInfov6> packageEntry : packageMap.entrySet()) {
+            String packageId = packageEntry.getKey();
+            Optional<Dependency> pnpmPackage = buildDependencyFromPackageEntry(packageEntry);
+            if (!pnpmPackage.isPresent()) {
+                logger.debug(String.format("Could not add package %s to the graph.", packageId));
+                continue;
+            }
+            
+            if (isRootPackage(packageId, rootPackageIds)) {
+                graphBuilder.addChildToRoot(pnpmPackage.get());
+            }
+
+            PnpmPackageInfov6 packageInfo = packageEntry.getValue();            
+            
+            if (!packageInfo.getDependencyType().isPresent() || dependencyTypeFilter.shouldInclude(packageInfo.getDependencyType().get())) {
+                for (Map.Entry<String, String> packageDependency : packageInfo.getDependencies().entrySet()) {
+                    addTransitiveDependencyToGraph(graphBuilder, linkedPackageResolver, reportingProjectPackagePath,
+                            pnpmPackage, packageDependency);
+                }
+                
+                if (dependencyTypeFilter.shouldInclude(PnpmDependencyType.DEV)) {
+                    for (Map.Entry<String, String> packageDependency : packageInfo.getDevDependencies().entrySet()) {
+                        addTransitiveDependencyToGraph(graphBuilder, linkedPackageResolver, reportingProjectPackagePath,
+                                pnpmPackage, packageDependency);
+                    }
+                }
+                
+                if (dependencyTypeFilter.shouldInclude(PnpmDependencyType.OPTIONAL)) {
+                    for (Map.Entry<String, String> packageDependency : packageInfo.getOptionalDependencies().entrySet()) {
+                        addTransitiveDependencyToGraph(graphBuilder, linkedPackageResolver, reportingProjectPackagePath,
+                                pnpmPackage, packageDependency);
+                    }
```

Comment:
We now have calls to obtain dependencies, devDependencies, and optionalDependencies. We'll add these to the graph if the user does not specifically filter them out. Previously v5 was only looking at dependencies.